cloud/linode/loadbalancers.go

```diff
@@ -229,6 +229,90 @@ func (l *loadbalancers) EnsureLoadBalancer(ctx context.Context, clusterName stri
 	return lbStatus, nil
 }
 
+func (l *loadbalancers) getNodeBalancerDeviceId(ctx context.Context, firewallID, nbID, page int) (int, bool, error) {
+	devices, err := l.client.ListFirewallDevices(ctx, firewallID, &linodego.ListOptions{PageSize: 500, PageOptions: &linodego.PageOptions{Page: page}})
+	if err != nil {
+		return 0, false, err
+	}
+
+	if len(devices) == 0 {
+		return 0, false, nil
+	}
+
+	for _, device := range devices {
+		if device.Entity.ID == nbID {
+			return device.ID, true, nil
+		}
+	}
+
+	return l.getNodeBalancerDeviceId(ctx, firewallID, nbID, page+1)
+}
+
+func (l *loadbalancers) updateNodeBalancerFirewall(ctx context.Context, service *v1.Service, nb *linodego.NodeBalancer) error {
+	var newFirewallID, existingFirewallID int
+	var err error
+	fwid, ok := service.GetAnnotations()[annLinodeCloudFirewallID]
+	if ok {
+		newFirewallID, err = strconv.Atoi(fwid)
+		if err != nil {
+			return err
+		}
+	}
+
+	// get the attached firewall
+	firewalls, err := l.client.ListNodeBalancerFirewalls(ctx, nb.ID, &linodego.ListOptions{})
+	if err != nil {
+		if !ok {
+			if err.Error() != "[404] Not Found" {
+				return err
+			} else {
+				return nil
+			}
+		}
+	}
+
+	if !ok && len(firewalls) == 0 {
+		return nil
+	}
+
+	if len(firewalls) != 0 {
+		existingFirewallID = firewalls[0].ID
+	}
+
+	if existingFirewallID != newFirewallID {
+		// remove the existing firewall
+
+		if existingFirewallID != 0 {
```

Comment:
when will existingFirewallID be 0?

Reply:
The value is zero when there is no Firewall assigned. Default value of a integer variable in golang is 0. The `existingFirewallID` variable would only be tampered if there's a firewall attached to nodeBalancer. If not, it would be 0

Reply:
ah I see the logic now. can you move the declaration for existingFirewallID to line 277 so its immediately obvious?